scheduled/cru-org-sync.js

```diff
@@ -0,0 +1,104 @@
+      headers: {
+        'Cache-Control': 'no-cache',
+        'User-Agent': 'Request-Promise'
+      },
+      json: true
+    }
+
+    return new Promise((resolve, reject) => {
+      request(assign({}, options, {url: `http://${hosts[0]}/bin/querybuilder.json`})).then(resolve, () => {
+        request(assign({}, options, {url: `http://${hosts[1]}/bin/querybuilder.json`})).then(resolve, reject)
+      })
+    })
+  }
+
+  /**
+   * Combine scored url with cruOrg Json
+   * @param record
+   * @param cruOrgJson
+   * @returns {*}
+   */
+  const scoreTransform = (record, cruOrgJson) => {
+    let recommendation = new Recommendation(record)
+
+    const path = recommendation.url.replace(prefixMatch, '/content/cru').replace('.html', '')
+    if (path.indexOf('.') === -1) {
+      let page = find(cruOrgJson.hits, {'jcr:path': path})
+      if (typeof page !== 'undefined') {
+        let [language, ...categories] = path.split('/').slice(4, -1)
+        if (typeof page['jcr:content'] !== 'undefined') {
+          let content = page['jcr:content']
+          if (typeof content['jcr:title'] !== 'undefined') {
+            recommendation.title = content['jcr:title'].trim()
+          }
+          if (typeof content['jcr:description'] !== 'undefined') {
+            recommendation.message = content['jcr:description'].replace(new RegExp('[\\r\\n]', 'gi'), ' ').trim()
+          } else if (typeof content['subtitle'] !== 'undefined') {
+            recommendation.message = content['subtitle'].replace(new RegExp('[\\r\\n]', 'gi'), ' ').trim()
+          }
+          if (typeof content['image'] !== 'undefined' && typeof content['image']['fileReference'] !== 'undefined') {
+            recommendation.thumbnail_url = recommendation.url.replace(/\.html$/i, '/_jcr_content/image.transform/CruHalf432x243/img.png')
+          }
+        }
+        // drop last category as it's the article name
+        categories.pop()
+        recommendation.categories = categories
+        recommendation.language = language
+        return recommendation.toJSON()
+      }
+    }
+  }
+
+  const transaction = await sequelize().transaction()
+  try {
+    // eslint-disable-next-line no-unused-vars
+    const [cruOrgJson, scores, _truncate] = await Promise.all([
+      getCruOrgJson(),
+      sequelize().query(scoresQuery, {
+        type: sequelize().QueryTypes.SELECT,
+        bind: [prefixMatch.source],
+        transaction: transaction
+      }),
+      Recommendation.truncate({transaction: transaction})
+    ])
+    const recommendations = []
+    forEach(scores, (score) => {
+      let rec = scoreTransform(score, cruOrgJson)
+      if (typeof rec !== 'undefined') {
+        recommendations.push(rec)
+      }
+    })
+    await Recommendation.bulkCreate(recommendations, {transaction: transaction})
+    await transaction.commit()
+    return `Successfully synced ${recommendations.length} recommendations.`
+  } catch (err) {
+    await transaction.rollback()
+    throw err
```

Comment:
Do we care to have a rollbar error log here?

Reply:
I'll add it in.